packages/@dcl/inspector/src/components/Tree/Tree.tsx

```diff
@@ -115,41 +118,50 @@ function Tree<T>(props: Props<T>) {
     onRemove(value)
   }
 
+  const ref = (node: HTMLDivElement | null) => drag(drop(node))
+
+  const controlsProps = {
+    enableAdd: enableAddChild,
+    enableEdit: enableRename,
+    enableRemove,
+    handleEdit: handleToggleEdit,
+    handleAdd: handleNewChild,
+    handleRemove: handleRemove
+  }
+
+  return (
+    <div ref={ref} className="main">
+      <div style={getLevelStyles(level)} className={selected ? 'selected' : ''}>
+        <span onClick={handleToggleExpand} style={getEditModeStyles(editMode)}>
+          <Arrow isOpen={open} />
+          <span>{label || id}</span>
+        </span>
+        {editMode && <Input value={label || ''} onCancel={quitEditMode} onSubmit={onChangeEditValue} />}
+        <Controls {...controlsProps} />
+      </div>
+      <TreeChildren {...props} />
+      {insertMode && <Input value="" onCancel={quitInsertMode} onSubmit={handleAddChild} />}
+    </div>
+  )
+}
+
+function Arrow({ isOpen }: { isOpen: boolean }) {
```

Comment:
We tend to keep one component per file. Making this a separate component seems a bit of an overkill to me tho, is it going to be used somewhere else? if not, then we could just inline this in the Tree component.